### PR TITLE
Update MSRV to 1.70 and remove lazy_static and once_cell

### PR DESCRIPTION
**Pull request by uklotzde**
Reduce some inconsistencies and remove obsolete crates.

### CODE REVIEW

**Review by orottier — APPROVED** [review]
Thanks, fixes #302 
Interesting performance improvement as well..
A MSRV of the current version is potentially disruptive. Let's wait at least for 1.71 to arrive (any time now, in pre-release)

File: src/node/audio_buffer_source.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use once_cell::sync::OnceCell;
+use std::cell::OnceCell;
```
Comment (uklotzde) [inline src/node/audio_buffer_source.rs:1]: Switching from `sync` to *non-sync* here might make a difference by avoiding a mutex.

File: src/node/stereo_panner.rs [hunk 2]
```diff
@@ -71,8 +71,10 @@ fn assert_valid_channel_count_mode(mode: ChannelCountMode) {
 #[inline(always)]
 fn get_stereo_gains(x: f32) -> [f32; 2] {
     let idx = (x * TABLE_LENGTH_BY_4_F32) as usize;
-    let gain_left = SINETABLE[idx + TABLE_LENGTH_BY_4_USIZE];
-    let gain_right = SINETABLE[idx];
+
+    let sine_table = sine_table();
```
Comment (uklotzde) [inline src/node/stereo_panner.rs:75]: Now the reference is only resolved once instead of twice per call.

Comment (uklotzde) [reply src/node/stereo_panner.rs]: Now the access to OnceLock only happens once during construction.

**Review by orottier — APPROVED** [review]

**Comment by uklotzde** [comment]
I don't see changes that would require bumping the MSRV to 1.71 right now: https://releases.rs/docs/1.71.0/

We could do this at any time when actually needed.

**Comment by uklotzde** [comment]
I have added a GitHub action for checking the MSRV. Let's see if this works.

**Pull request merged**